### PR TITLE
Fix minor type format issue. (PHP, Go)

### PR DESCRIPTION
**Pull request by SilasKenneth**
## Overview

Captures more type formats for string values.
 - Boolean.
 - Double.
 - Duration.
It also simplifies the `evaluateStringProperty` method in `SnippetCodeGraph.cs`

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 1]
```diff
@@ -272,6 +272,9 @@ private static void WriteCodeProperty(string propertyAssignment, StringBuilder b
                     builder.AppendLine($"{indentManager.GetIndent()}{NormalizeJsonName(child.Name.ToFirstCharacterLowerCase())} := []byte(\"{child.Value.ToFirstCharacterLowerCase()}\")");
                     builder.AppendLine($"{propertyAssignment}.Set{propertyName.ToFirstCharacterUpperCase()}(&{propertyName}) ");
                     break;
+                case PropertyType.Duration:
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:275]: just call parse iso duration https://github.com/microsoft/kiota-abstractions-go/blob/1ecd925f435fb1aaf0174576a0ef222044c7217a/serialization/iso_duration.go#L85

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 2]
```diff
@@ -22,6 +23,20 @@ public record SnippetCodeGraph
 
         private static readonly CodeProperty EMPTY_PROPERTY = new() { Name = null, Value = null, Children = null, PropertyType = PropertyType.Default };
 
+        private static Dictionary<string, PropertyType> _formatPropertyTypes = new ()
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:26]:
```suggestion
        private static Dictionary<string, PropertyType> _formatPropertyTypes = new (StringComparer.OrdinalIgnoreCase)
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs (hunk 2 above)
Comment (SilasKenneth) [reply CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:26]: I think given that the keys are in Lowercase and I'm doing case conversion on line https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/e948a26de6cc2c7ecf8d16f5ccc6c18dd3472aec/CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs#L281
we might not need to add the suggestion.

Comment (baywet) [reply CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:26]: it is generally better to use the right comparer than forcing a casing before comparison.
https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs (hunk 1 above)
Comment (SilasKenneth) [reply CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:275]: Done!

**Review by baywet — APPROVED** [review]

**Pull request merged**